input/postgres/backends.go

```diff
@@ -73,8 +96,71 @@ func GetBackends(logger *util.Logger, db *sql.DB, postgresVersion state.Postgres
 			row.BackendType.String = strings.ToValidUTF8(row.BackendType.String, "")
 		}
 
+		// Store blocking graph to calculate blocking info later on
+		for _, i := range row.BlockedByPids {
+			// "from" is blocking "to"
+			blockingEdge := Edge{from: i, to: int64(row.Pid)}
+			blockingGraph = append(blockingGraph, blockingEdge)
+		}
+
 		activities = append(activities, row)
 	}
 
+	for i, activity := range activities {
+		pid := int64(activity.Pid)
+		activities[i].BlockingPids = findBlocking(blockingGraph, pid)
+		activities[i].IndirectlyBlockingPids = findIndirectlyBlocking(blockingGraph, activities[i].BlockingPids)
+		activities[i].IndirectlyBlockedByPids = findIndirectlyBlockedBy(blockingGraph, activities[i].BlockedByPids)
+	}
+
 	return activities, nil
 }
+
+func findBlocking(graph []Edge, pid int64) []int64 {
```

Comment:
We could move this logic into the `util` folder for better abstraction (together with the test), since its fairly independent. That assumes we keep the logic in the collector.